functional/streams.py

```diff
@@ -47,6 +50,17 @@ def __call__(self, *args, no_wrap=None, **kwargs):
             args, engine, no_wrap=default_value(no_wrap, self.no_wrap, False)
         )
 
+    def chain(self, *args, no_wrap=None, **kwargs):
+        """
+        Merge given the iterators firstly, then new the seq.
+        """
+        for arg in args:
+            if not isinstance(arg, Iterable):
+                raise TypeError("The type of arg should be iterator.")
+
+        merged = list(chain.from_iterable(args))
+        return self(*merged, no_wrap=no_wrap, **kwargs)
```

Comment:
Could you test without chain?
```suggestion
        return self(chain.from_iterable(args), no_wrap=no_wrap, **kwargs)
```
I think the easiest way to test laziness would be to use a function yielding elements.

Reply:
But, in `_parse_args`, args would be `list` finally.

Reply:
No? I don't use * to destructure, so len(args) == 1 and it should keep the iterator.